plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/MavenServerManager.java

```diff
@@ -216,7 +218,16 @@ public JavaParameters buildMavenServerParameters() {
 
     parameters.getClassPath().addAll(classPath);
 
-    parameters.getVmParameters().add("-Xmx512m");
+    String vmArguments =
+        MoreObjects.firstNonNull(
+            System.getenv("CHE_WORKSPACE_MAVEN_SERVER_JAVA_OPTIONS"),
+            MoreObjects.firstNonNull(
+                System.getenv("CHE_WORKSPACE_MAVEN_SERVER_JAVA_OPTIONS_DEFAULT"), "-Xmx128m"));
+
+    Arrays.stream(vmArguments.split(" "))
```

Comment:
It is better to do this in constructor instead of doing it on each call of getExecutor method of RmiClient